src/pack.c

```diff
@@ -1185,6 +1186,11 @@ static git_off_t nth_packed_object_offset(const struct git_pack_file *p, uint32_
 		if (!(off & 0x80000000))
 			return off;
 		index += p->num_objects * 4 + (off & 0x7fffffff) * 8;
+
+		/* Make sure we're not being sent out of bounds */
+		if (index + 4 >= end)
```

Comment:
Anyting past `end - 2*GIT_OID_RAWSZ` would be out of bounds, just wouldn't cause a segfault, so this is being generous. But we're reading 4 bytes so this is actually not enough, we need to check if `index + 8 > end`.